app/models/series_list.rb

```diff
@@ -1,14 +1,19 @@
+require 'set'
+
 class SeriesList
-  def initialize(pairs = nil)
-    pairs ||= Hash[
-      RSolr.connect(url: 'http://localhost:8983/solr/')
-      .get('select', params: {
-             'facet.field' => 'series_title',
-             'facet' => 'true',
-             'rows' => '0',
-             'f.series_title.facet.limit' => -1
-           }
-          )['facet_counts']['facet_fields']['series_title'].in_groups_of(2)]
+  def initialize(series_all = nil, series_online = nil) # Optional params only for tests
+    series_all ||= SeriesList.series_facet(false)
+    series_online ||= SeriesList.series_facet(true)
+
```

Comment:
@afred: hmm: `SeriesList.series_facet` could go above, instead of nil... I guess putting it all in the method signature seemed a little hard to read, but not a strong preference.